Packs/QRadar/Integrations/QRadar_v2/QRadar_v2.py

```diff
@@ -0,0 +1,1961 @@
+    """
+    Look for offenses in QRadar using an increasing search window until a fetchable offense is found
+    """
+    raw_offenses = []
+    fetch_query = ""
+    lim_id = None
+    latest_offense_fnd = False
+    while not latest_offense_fnd:
+        end_offense_id = int(start_offense_id) + client.offenses_per_fetch + 1
+        fetch_query = "id>{0} AND id<{1} {2}".format(
+            start_offense_id,
+            end_offense_id,
+            "AND ({})".format(user_query) if user_query else "",
+        )
+        print_debug_msg(f"Fetching {fetch_query}.")
+        raw_offenses = client.get_offenses(
+            _range="0-{0}".format(client.offenses_per_fetch - 1), _filter=fetch_query
+        )
+        if raw_offenses:
+            latest_offense_fnd = True
+        else:
+            if not lim_id:
+                # set fetch upper limit
+                lim_offense = client.get_offenses(_range="0-0")
+                if not lim_offense:
+                    raise Exception(
+                        "No offenses could be fetched, please make sure there are offenses available for this user."
+                    )
+                lim_id = lim_offense[0]["id"]  # if there's no id, raise exception
+            if lim_id >= end_offense_id:  # increment the search until we reach limit
+                start_offense_id += client.offenses_per_fetch
+            else:
+                latest_offense_fnd = True
+    return raw_offenses, fetch_query
+
+
+def fetch_incidents_long_running_samples():
+    last_run = get_integration_context(SYNC_CONTEXT)
+    return last_run.get("samples", [])  # type: ignore [attr-defined]
+
+
+def is_reset_triggered(lock, handle_reset=False):
+    """
+    Returns if reset signal is set. If handle_reset=True, will also reset the integration context
+    """
+    if lock.acquire(timeout=LOCK_WAIT_TIME):
+        ctx = get_integration_context(SYNC_CONTEXT)
+        if RESET_KEY in ctx:
+            if handle_reset:
+                print_debug_msg("Reset fetch-incidents.")
+                set_integration_context(
+                    {"samples": ctx.get("samples", [])}, sync=SYNC_CONTEXT
+                )
+            lock.release()
+            return True
+        lock.release()
+    return False
+
+
+def fetch_incidents_long_running_events(
+    client: QRadarClient,
+    user_query,
+    full_enrich,
+    fetch_mode,
+    events_columns,
+    events_limit,
+):
+    last_run = get_integration_context(SYNC_CONTEXT)
+    offense_id = last_run["id"] if last_run and "id" in last_run else 0
+
+    raw_offenses = fetch_raw_offenses(client, offense_id, user_query)
+
+    if len(raw_offenses) == 0:
+        return
+    if isinstance(raw_offenses, list):
+        raw_offenses.reverse()
+    for offense in raw_offenses:
+        offense_id = max(offense_id, offense["id"])
+    enriched_offenses = []
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
If I understand correctly, I think you are creating a new thread pool each time which is less optimal. Better to use a global thread pool and reuse it.